src/java/fr/paris/lutece/plugins/searchstats/service/SearchStatsService.java

```diff
@@ -0,0 +1,143 @@
+        }
+
+    }
+
+    /**
+     * @param nYear
+     *            nMonth nDay nHour The limit Year / Month / Day / Hour
+     * @param listQueryRecords
+     *            The list of Query Records
+     * @param nLimit
+     *            The limit number of queries
+     * @return The list of the QueryRecords
+     */
+    public List<TermRecord> getMapSearchStatsAfterdate( int nYear, int nMonth, int nDay, int nHour, List<QueryRecord> listQueryRecords, int nLimit )
+    {
+        List<QueryRecord> listQueryRecordAfterDate = new ArrayList<>();
+
+        TimedRecord timeReport = new TimedRecord( );
+        timeReport.setYear( nYear );
+        timeReport.setMonth( nMonth );
+        timeReport.setDay( nDay );
+        timeReport.setHour( nHour );
+
+        for ( QueryRecord queryRecord : listQueryRecords )
+        {
+            if ( checkAfterDate( timeReport, queryRecord ) )
+            {
+            	listQueryRecordAfterDate.add( queryRecord );
+            }
+        }
+        List<TermRecord> listTermRecords = TermRecordService.getTopTerms(listQueryRecordAfterDate);
+        return listTermRecords;
+    }
+
+    /**
+     * @return The timeRecord
+     */
+    public static TimedRecord getIntervalTimeRecord( )
+    {
+        TimedRecord timedRecord = new TimedRecord( );
+        String strNombreSecondes = DatastoreService.getInstanceDataValue( STATS_DAEMON_INTERVAL, DEFAULT_INTERVAL );
+        int nStrNombreSecondes = Integer.parseInt( strNombreSecondes );
+        // Converti les secondes en Y/M/D/H
+        int nYear = ( nStrNombreSecondes / 31536000 );
+        int nMonth = ( ( nStrNombreSecondes - ( nYear * 31536000 ) ) / 2628000 );
+        int nDay = ( ( nStrNombreSecondes - ( nYear * 31536000 + nMonth * 2628000 ) ) / 86400 );
+        int nHour = ( ( nStrNombreSecondes - ( nYear * 31536000 + nMonth * 2628000 + nDay * 86400 ) ) / 3600 );
+        // Set le timeRecord
+        timedRecord.setYear( nYear );
+        timedRecord.setMonth( nMonth );
+        timedRecord.setDay( nDay );
+        timedRecord.setHour( nHour );
+        return timedRecord;
+    }
+
+    /**
+     *
+     * @param listQueryRecords
+     * @param nLimit
+     * @return The list of the QueryRecords after the last daemon
+     */
+    public List<TermRecord> getMapSearchStatsAfterLastDaemon( List<QueryRecord> listQueryRecords, int nLimit )
+    {
+        TimedRecord timeReport = getIntervalTimeRecord( );
+        if ( timeReport.getYear( ) == 0 && timeReport.getMonth( ) == 0 && timeReport.getDay( ) == 0 )
+        {
+            timeReport.setDay( 1 );
+        }
+        return getMapSearchStatsAfterdate( timeReport.getYear( ), timeReport.getMonth( ), timeReport.getDay( ), timeReport.getHour( ), listQueryRecords, nLimit );
+    }
+
+    /**
+     * @param timeReport
+     *            The time from which the query can be taken into account
```

Comment:
It seems to me that this represents a duration (which you then remove from current time to get a new time)